Fix compiler warnings in lwIP build

The lwip build is currently suffering from a lot of compiler warnings (mainly) due to unused parameters:

Just taking net.c as an example:

/libcoap/src/net.c:711:71: warning: unused parameter 'node' [-Wunused-parameter]
/libcoap/src/net.c:1108:27: warning: unused parameter 'ctx' [-Wunused-parameter]
/libcoap/src/net.c:1108:44: warning: unused parameter 'now' [-Wunused-parameter]
/libcoap/src/net.c:2577:12: warning: variable 'us' set but not used [-Wunused-but-set-variable]
/libcoap/src/net.c:2803:29: warning: format '%d' expects argument of type 'int', but argument 2 has type 'coap_tick_t' {aka 'long unsigned int'} [-Wformat=]

These issues should be fixed to clean the build output. For unused parameters, make it explicit they are not used e.g. by
(void) node;
and so on.

--- a/examples/lwip/Makefile
+++ b/examples/lwip/Makefile
@@ -112,8 +112,6 @@ C_COAP_OBJ =$(patsubst %.c,lib-client/%.o,$(COAP_SRC))
 S_COAP_OBJ =$(patsubst %.c,lib-server/%.o,$(COAP_SRC))
 
 CFLAGS += -g3 -Wall -Wextra -pedantic -O0
-# not sorted out yet
-CFLAGS += -Wno-unused-parameter
 
 CFLAGS += -Ilwip/src/include/ -Ilwip/src/include/ipv4/ \
 	  -Ilwip-contrib/ports/unix/port/include/ \

--- a/examples/lwip/client-coap.c
+++ b/examples/lwip/client-coap.c
@@ -32,6 +32,9 @@ message_handler(coap_session_t *session,
   size_t offset;
   size_t total;
 
+  (void)session;
+  (void)sent;
+  (void)id;
   if (coap_get_data_large(received, &len, &data, &offset, &total)) {
     printf("%*.*s", (int)len, (int)len, (const char*)data);
     if (len + offset == total) {

--- a/examples/lwip/server-coap.c
+++ b/examples/lwip/server-coap.c
@@ -34,6 +34,9 @@ hnd_get_time(coap_resource_t *resource, coap_session_t  *session,
   coap_tick_t now;
   coap_tick_t t;
 
+  (void)resource;
+  (void)session;
+  (void)request;
   /* FIXME: return time, e.g. in human-readable by default and ticks
    * when query ?ticks is given. */
 

--- a/include/coap3/mem.h
+++ b/include/coap3/mem.h
@@ -131,10 +131,12 @@ COAP_STATIC_INLINE void coap_memory_init(void) {}
  * made conditional.
  */
 COAP_STATIC_INLINE void *coap_malloc(size_t size) {
+  (void)size;
   LWIP_ASSERT("coap_malloc must not be used in lwIP", 0);
 }
 
 COAP_STATIC_INLINE void coap_free(void *pointer) {
+  (void)pointer;
   LWIP_ASSERT("coap_free must not be used in lwIP", 0);
 }
 

--- a/src/coap_debug.c
+++ b/src/coap_debug.c
@@ -289,6 +289,8 @@ coap_print_addr(const coap_address_t *addr, unsigned char *buf, size_t len) {
 
   return strlen((char *)p);
 # else /* WITH_CONTIKI */
+  (void)addr;
+  (void)len;
   /* TODO: output addresses manually */
 #   warning "inet_ntop() not available, network addresses will not be included in debug output"
 # endif /* WITH_CONTIKI */

--- a/src/coap_io_lwip.c
+++ b/src/coap_io_lwip.c
@@ -347,6 +347,11 @@ coap_socket_send_pdu(coap_socket_t *sock, coap_session_t *session,
 ssize_t
 coap_socket_send(coap_socket_t *sock, coap_session_t *session,
   const uint8_t *data, size_t data_len ) {
+
+  (void)sock;
+  (void)session;
+  (void)data;
+  (void)data_len;
   /* Not implemented, use coap_socket_send_pdu instead */
   return -1;
 }
@@ -355,6 +360,9 @@ int
 coap_socket_bind_udp(coap_socket_t *sock,
   const coap_address_t *listen_addr,
   coap_address_t *bound_addr) {
+  (void)sock;
+  (void)listen_addr;
+  (void)bound_addr;
   return 0;
 }
 
@@ -369,6 +377,10 @@ coap_socket_connect_udp(coap_socket_t *sock,
   err_t err;
   struct udp_pcb *pcb;
 
+  (void)local_if;
+  (void)default_port;
+  (void)local_addr;
+  (void)remote_addr;
   pcb = udp_new();
 
   if (!pcb) {
@@ -397,6 +409,7 @@ coap_socket_connect_udp(coap_socket_t *sock,
 }
 #endif /* ! COAP_CLIENT_SUPPORT */
 
+#if ! COAP_DISABLE_TCP
 int
 coap_socket_connect_tcp1(coap_socket_t *sock,
                          const coap_address_t *local_if,
@@ -404,6 +417,12 @@ coap_socket_connect_tcp1(coap_socket_t *sock,
                          int default_port,
                          coap_address_t *local_addr,
                          coap_address_t *remote_addr) {
+  (void)sock;
+  (void)local_if;
+  (void)server;
+  (void)default_port;
+  (void)local_addr;
+  (void)remote_addr;
   return 0;
 }
 
@@ -411,6 +430,9 @@ int
 coap_socket_connect_tcp2(coap_socket_t *sock,
                          coap_address_t *local_addr,
                          coap_address_t *remote_addr) {
+  (void)sock;
+  (void)local_addr;
+  (void)remote_addr;
   return 0;
 }
 
@@ -418,6 +440,9 @@ int
 coap_socket_bind_tcp(coap_socket_t *sock,
                      const coap_address_t *listen_addr,
                      coap_address_t *bound_addr) {
+  (void)sock;
+  (void)listen_addr;
+  (void)bound_addr;
   return 0;
 }
 
@@ -426,16 +451,27 @@ coap_socket_accept_tcp(coap_socket_t *server,
                         coap_socket_t *new_client,
                         coap_address_t *local_addr,
                         coap_address_t *remote_addr) {
+  (void)server;
+  (void)new_client;
+  (void)local_addr;
+  (void)remote_addr;
   return 0;
 }
+#endif /* !COAP_DISABLE_TCP */
 
 ssize_t
 coap_socket_write(coap_socket_t *sock, const uint8_t *data, size_t data_len) {
+  (void)sock;
+  (void)data;
+  (void)data_len;
   return -1;
 }
 
 ssize_t
 coap_socket_read(coap_socket_t *sock, uint8_t *data, size_t data_len) {
+  (void)sock;
+  (void)data;
+  (void)data_len;
   return -1;
 }
 

--- a/src/coap_session.c
+++ b/src/coap_session.c
@@ -329,6 +329,7 @@ static size_t
 coap_session_max_pdu_size_internal(const coap_session_t *session,
                                    size_t max_with_header) {
 #if COAP_DISABLE_TCP
+  (void)session;
   return max_with_header > 4 ? max_with_header - 4 : 0;
 #else /* !COAP_DISABLE_TCP */
   if (COAP_PROTO_NOT_RELIABLE(session->proto))

